test/lint/README.md

```diff
@@ -37,6 +37,7 @@ Then you can use:
 | [`lint-python-dead-code.py`](/test/lint/lint-python-dead-code.py) | [vulture](https://github.com/jendrikseipp/vulture)
 | [`lint-shell.py`](/test/lint/lint-shell.py) | [ShellCheck](https://github.com/koalaman/shellcheck)
 | [`lint-spelling.py`](/test/lint/lint-spelling.py) | [codespell](https://github.com/codespell-project/codespell)
+| markdown link check | [mlc](https://github.com/becheran/mlc)
```

Comment:
Sure it could be, but as I see it would need us to write our own markdown parser and link checker.

I don't have the appetite for that myself, hence this approach.

Also see https://github.com/bitcoin/bitcoin/pull/29965 which may (I didn't fully check yet) permit you to run this lint individually in the future :)